docs/fft.rst

```diff
@@ -0,0 +1,170 @@
+    >>> print(outarr)
+
+or (this one is an inverse FFT)::
+
+    >>> import numpy as np
+    >>> from pycbc import types
+    >>> from pycbc import fft
+    >>> inarr = types.Array(np.ones([33], dtype=np.complex64))
+    >>> outarr = types.Array(np.zeros([64], dtype=np.float32))
+    >>> fft.ifft(inarr, outarr)
+    >>> print(outarr)
+
+This will work the pycbc Timeseries and Frequencyseries as well. Except you
+must FFT a TimeSeries to a FrequencySeries or IFFT a FrequencySeries to a
+TimeSeries. In this case the time and frequency spacing must also be
+consistent. For this reason we provide convenience functions that use the
+function API, but figure out these details, and create the output array, for
+you. As an example::
+
+    >>> import numpy as np
+    >>> from pycbc import types
+    >>> inarr = types.TimeSeries(np.ones([64], dtype=np.float64), delta_t=1./64.)
+    >>> outarr = inarr.to_frequencyseries()
+
+or::
+
+    >>> import numpy as np
+    >>> from pycbc import types
+    >>> inarr = types.FrequencySeries(np.ones([33], dtype=np.complex128), delta_f=1.)
+    >>> outarr = inarr.to_timeseries()
+
+
+=====================
+Using the class-based API
+=====================
+
+The PyCBC class-based API should be used if you care about performance. If you
+are performing FFTs many times, with inputs that are the same size each time,
+using this will offer significance perfommance improvement. 
+
+Here's how to use this::
+
+    >>> from pycbc import fft
+    >>> fft_class = pycbc.fft.FFT(inarr, outarr)
+    >>> fft_class.execute()
+    >>> outarr *= inarr._delta_t # ONLY IF inarr is a TimeSeries
+    >>> outarr *= inarr._delta_f # ONLY IF inarr is a FrequencySeries
+
+Or for an inverse FFT::
+
+    >>> from pycbc import fft
+    >>> ifft_class = pycbc.fft.IFFT(inarr, outarr)
+    >>> ifft_class.execute()
+    >>> outarr *= inarr._delta_t # ONLY IF inarr is a TimeSeries
+    >>> outarr *= inarr._delta_f # ONLY IF inarr is a FrequencySeries
+
+The idea would be that the `fft_class` or `ifft_class` would only be created
+*once* and the execute command called many times. You would change the contents
+of inarr before each call and outarr will update when execute is run. After
+creating the FFT class *do not* reassign inarr, but instead set values. So::
+
+    >>> fft_class = pycbc.fft.FFT(inarr, outarr)
+    >>> inarr = types.TimeSeries(np.ones([64], dtype=np.float64), delta_t=1./64.)
+    >>> fft_class.execute()
+
+would not work! Instead do::
+
+    >>> fft_class = pycbc.fft.FFT(inarr, outarr)
+    >>> inarr[:] = np.ones([64], dtype=np.float64)[:]
+    >>> fft_class.execute()
+
+
+====================
+Choosing a specific backend
+====================
+
+If you want to choose a specific backend, you can see what is available with::
+
+    >>> from pycbc.fft import backend_support
+    >>> backend_support.get_backend_names()
+
+and then do::
+
+    >>> from pycbc.fft import backend_support
+    >>> backend_support.set_backend('mkl')
+
```

Comment:
Have you tested this?  I would only expect
```
backend_support.set_backend(['mkl'])
```
to work.  That is, `set_backend` expects a priority-ordered list.  Not that it's unreasonable to support a single entry, but I don't think the code works that way right now? I missed it if you have changed that as part of this PR.

Reply:
Yes, you're right, this is a typo on my part. It should be what you say, I'll fix this.